File: my_py_pkg/my_py_pkg/my_first_node.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node


class myNode(Node):
    def __init__(self):
        super().__init__(node_name="py_test")
        self.counter_ = 0
        self.logger = self.get_logger()
        self.logger.info("Node 1 Started !!!!!!!")
        self.create_timer(0.5, self.timer_callback)

    def timer_callback(self):
        self.counter_ += 1
        self.logger.info(f"Hello World {self.counter_}")


def main(args=None):
    rclpy.init(args=args)
    node = myNode()
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == "__main__":
    main()

File: my_cpp_pkg/src/my_first_node.cpp
#include "rclcpp/rclcpp.hpp"

class MyNode : public rclcpp::Node
{
public:
    MyNode() 
        : Node("cpp_test"), 
        counter_(0), 
        logger_(this->get_logger())
    {
        RCLCPP_INFO(logger_, "hello cpp node OOP");

        timer_ = this->create_wall_timer(std::chrono::seconds(1), std::bind(&MyNode::timerCallback, this));
    }

private: 
    void timerCallback()
    {
        counter_++;
        RCLCPP_INFO(logger_, "Hello World! %d", counter_);
    }

    rclcpp::TimerBase::SharedPtr timer_;
    int counter_;
    rclcpp::Logger logger_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<MyNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}